### PR TITLE
added robot configuration lookup service

### PR DESCRIPTION
**Pull request by younesr1**
This PR defines some joint configurations for the robot. I found these configurations through sim and they looked good. the configurations defined will likely need to be adjusted on the robot, especially the arm configurations.

Here is a picture of the robot's home configuration:

Screenshot from 2023-01-29 16-34-54
Screenshot from 2023-01-29 16-31-36

### CODE REVIEW

File: nightingale_msgs/srv/RobotConfigurationLookup.srv [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# body_part is one of {left_arm,right_arm,torso,head}
+int8 LEFT_ARM=0
+int8 RIGHT_ARM=1
+int8 TORSO=2
+int8 HEAD=3
+int8 body_part
+---
+sensor_msgs/JointState jnt_state
```
Comment (FelixKHWong) [inline nightingale_msgs/srv/RobotConfigurationLookup.srv:8]: I think you've got a slight spelling mistake here in the variable name "int_state"

**Review by FelixKHWong — CHANGES_REQUESTED** [review]
Just to confirm: the new service is used to query for the states of the robot so we can figure out the required motions needed to perform specific actions? So if we want to move the end effector to position X we can easily calculate the necessary cartesian movement from the current position?

File: nightingale_msgs/srv/RobotConfigurationLookup.srv (hunk 1 above)
Comment (younesr1) [reply nightingale_msgs/srv/RobotConfigurationLookup.srv:8]: jnt_state is short for joint_state, which is the type of the msg. "jnt" is a common abbreviation for "joint" used in many places in the kinova repo.

Example is line 385 in movo_jtas.py
<for jnt in joint_names>

**Comment by younesr1** [comment]
> Just to confirm: the new service is used to query for the states of the robot so we can figure out the required motions needed to perform specific actions? So if we want to move the end effector to position X we can easily calculate the necessary cartesian movement from the current position?

Not quite. This PR defines some joint configurations that the robot should be in in certain situations. For example, when the robot boots up, the gimbal should face ahead and the torso should be at a predetermined height, so the mission planner will (eventually, this is not an urgent need and doesnt need to exist before saturday) query the target torso height from this server and then send a command to some other node which make the torso move.

Another example is with the arms. The service server i've created defines our own "home" configuration for the arms. So the mission-planner can query this service for the arm home configurations and then ask Ben's code to achieve these configurations.

When I say configuration i meant the specific joint angles of whatever sub-component (ie. the head). The head configuration has 2 numbers (pan & tilt)

File: nightingale_utils/configs/robot_configuration_lookup.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+robot_configurations: {
```
Comment (le2tang) [inline nightingale_utils/configs/robot_configuration_lookup.yaml]: Can we change this to `joint_configurations`. I expect to have a `cartesian_configurations` that does something similar to what you have here, except in cartesian space.

File: nightingale_utils/src/utils.py [hunk 3]
```diff
@@ -56,8 +58,35 @@ def room_pose_lookup(self, req):
...
+class RobotConfigurationService:
+    def __init__(self):
+        self.ss = rospy.Service(
+            "/nightingale/robot_configuration_lookup",
+            RobotConfigurationLookup,
+            self.robot_configuration_lookup,
+        )
+        self.configurations = rospy.get_param("/nightingale_utils/robot_configurations")
+
+    def robot_configuration_lookup(self, req):
```
Comment (le2tang) [inline nightingale_utils/src/utils.py:72]: Nit: Add a check to make sure all the lookup arguments are defined in self.configurations

**Review by le2tang — DISMISSED** [review]

File: nightingale_utils/configs/robot_configuration_lookup.yaml (hunk 2 above)
Comment (younesr1) [reply nightingale_utils/configs/robot_configuration_lookup.yaml]: how about "home_joint_configurations"? these values represent "home".

**Review by bdschnapp — DISMISSED** [review]

**Review by le2tang — DISMISSED** [review]

**Review by FelixKHWong — DISMISSED** [review]

**Review by le2tang — DISMISSED** [review]

**Review by le2tang — APPROVED** [review]

**Pull request merged**